.azure/pipelines/Phonebook.preview.yml

```diff
@@ -0,0 +1,213 @@
+          inline: |
+            apiVersion: v1
+            kind: Namespace
+            metadata:
+              name: $(namespace)
+
+      - task: HelmDeploy@0
+        displayName: 'helm init'
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: 'Phonebook Kubernetes Demo Cluster'
+          namespace: $(namespace)
+          command: init
+          arguments: '--node-selectors "beta.kubernetes.io/os"="linux"'
+          tillerNamespace: $(namespace)
+
+      - task: HelmDeploy@0
+        displayName: 'helm upgrade'
+        inputs:
+          connectionType: 'Kubernetes Service Connection'
+          kubernetesServiceConnection: 'Phonebook Kubernetes Demo Cluster'
+          namespace: $(namespace)
+          command: upgrade
+          chartType: FilePath
+          chartPath: '$(Pipeline.Workspace)/$(helm_artifact_name)/phonebook-0.1.0.tgz'
+          releaseName: phonebook-$(namespace)
+          overrideValues: 'frontend.image.tag=$(image_tag),frontend.image.repository=danielhabenicht,frontend.image.name=phonebook,traefik.kubernetes.namespaces={$(namespace)},traefik.loadBalancerIP=$(public_ip),host=$(public_ip)'
+          valueFile: 'Phonebook.Demo/values.yml'
+          recreate: true
+          tillerNamespace: $(namespace)
+
+      - task: Kubernetes@1
+        displayName: 'kubectl create assets'
+        inputs:
+          kubernetesServiceEndpoint: 'Phonebook Kubernetes Demo Cluster'
+          namespace: 'kube-public'
+          command: apply
+          arguments: '-f https://raw.githubusercontent.com/T-Systems-MMS/phonebook/master/Phonebook.Demo/Phonebook.Assets.yml'
+
+      - task: Kubernetes@1
+        displayName: 'kubectl create MockBackend'      
+        inputs:
+          kubernetesServiceEndpoint: 'Phonebook Kubernetes Demo Cluster'
+          namespace: 'kube-public'
+          command: apply
+          arguments: '-f https://raw.githubusercontent.com/T-Systems-MMS/phonebook/master/Phonebook.Demo/Phonebook.MockBackend.yml'
+
+    - job: notify
+      displayName: 'Notify Github '
+      pool: server
+      dependsOn: deploy_preview
+      variables: 
+        public_ip: $[ dependencies.deploy_preview.outputs['get_public_ip.public_ip'] ]
+      steps:
+      - task: InvokeRESTAPI@1
+        inputs:
+          connectionType: 'connectedServiceName'
+          serviceConnection: 'Github API Test'
+          method: 'POST'
+          headers: |
+            {
+            "Content-Type":"application/json"
+            }
+          body: |
+            {
+              "body": "Preview Environment ready at https://$(public_ip)"
+            }
+          urlSuffix: 'repos/T-Systems-MMS/phonebook/issues/$(System.PullRequest.PullRequestNumber)/comments'
+          waitForCompletion: 'false'
+        
+
+- stage: delete
+  displayName: 'Delete Preview'
+  dependsOn: deploy
+  condition: or(failed(), succeeded())
+  jobs:
+  - job: delay
```

Comment:
@paule96 Vielleicht hast du hier noch eine Idee: Ich möchte das der Task nur ausgeführt wird wenn die vorherige **stage** den Status `succeeded` hat. Wenn nicht soll die Wartezeit übersprungen werden.

Reply:
du hast da in der Condition schon succeded drin stehen. Also geht.

Reply:
Jo nach den wir das mit den Jobs umgestellt haben